3.2 backports July 2023 edition

vadz opened the pull request.

swt2c committed:
Add support for building WebView with libwebkit2gtk-4.1

libwebkit2gtk-4.1 has the same API as libwebkit2gtk-4.0, except that the
former links with libsoup-3.0 and the latter links with libsoup-2.4.

Fixes #23630.

(cherry picked from commit 1b8664426603376b68f8ca3c54de97ec630e5940)

swt2c committed:
Switch Fedora CI to use libwebkit2gtk-4.1

(cherry picked from commit 411e183cd0c3f9961083d94d9b2f09652a401720)

Rezakarimi66 committed:
Invalidate wxOwnerDrawnComboBox best size after adding new items

The previously computed best size may become invalid after inserting new
items, so forget it.

See #23681.

(cherry picked from commit a9d2a140900f8214f7d724889fcf9f06c27db7d2)

sz5000 committed:
Fix possible crash in wxWindowsPrintNativeData initialization

If the printer name is specified, avoid fall back to the default printer
via the use of PRINTDLG, as the default printer DEVMODE may be
incompatible with that of the printer we're actually going to use and
could possible result in a crash if their size differs.

Use a temporary printer if necessary to retrieve the correct DEVMODE for
the given printer name.

See #23685.

(cherry picked from commit e35257d2b6fdc301c1f76f514350191bedebf0bd)

swt2c committed:
Read Linux distribution info from os-release file

The Linux distribution community has somewhat deprecated the lsb_release
utility and has standardized on a new file, os-release, that can be
simply parsed to get the same information.  Attempt to read this file in
/etc/os-release, then /usr/lib/os-release, and finally, fall back to
using the lsb_release utility if neither of those files are found.

See: https://www.freedesktop.org/software/systemd/man/os-release.html
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2184391

See #23712.

(cherry picked from commit aef7df6c9f44f751d97f4f6519ae6e5c3b81019d)

PBfordev committed:
Add MSVS 2022 to supported compilers in MSW binaries docs

See #23714.

PBfordev committed:
Document that RUNTIME_LIBS works only when building with MSVC

Mention that RUNTIME_LIBS=static works only when building
wxWidgets with Microsoft Visual C++.

See #23715.

vadz committed:
Really fix wxProcessDialog size when the message length changes

Ensure that we only increase the dialog width but never decrease it to
prevent the dialog from jumping annoyingly to different size with each
label change.

The previous attempt to do this in 4101849b4d (Grow
wxGenericProgressDialog automatically if needed, 2017-10-27) didn't work
because it seemingly didn't take into account the fact that wxStaticText
width changes when its label changes and so calling SetLabel() could
make it smaller, meaning that the next update would call Fit() again.

So fix this by preventing the control from changing its size by using
wxST_NO_AUTORESIZE and then only resizing it manually if it really needs
more space.

See #10624, #23727.

(cherry picked from commit 97546a6f0cf941ce94b42a27b23f51e224c8e54e)

vadz committed:
Merge branch 'webkit2_support_4.1' into 3.2

Add support for building WebView with libwebkit2gtk-4.1.

See #23713.

vadz committed:
Fix deleting selected item in wxGenericListCtrl

This didn't update the number of selected items, which resulted in other
problems later on, including accessing items with invalid index which
could lead to a crash.

This is a modified version of d6dd0ebb6e25ac4fbde7f3cca2074be42a54c10d
from master.

See #23729.

swt2c commented:
@vadz can you please add the out-of-tree samples and os-release changes to the backport queue as well.  :-)

out-of-tree commits: 0a1b92f (branch merge), b6ea233, 8b83e45 (CI)
os-release: aef7df6

vadz committed:
Fix wxGLCanvas performance when using Wayland

See #23554.

(cherry picked from commit 414140113c19b100986f340e5875e209fca1fac1)

vadz committed:
Fix wxMSW wxComboCtrl build with pre-C++11 compilers

Replace a lambda used in c445798fa8 (Fix using wxComboCtrl popup inside
AUI floating panes, 2023-03-31) with a separate function.

vadz commented:
Sorry @swt2c I'm getting some weird changes to 3rd party libs (PNG etc) when trying to backport the samples-related stuff and rebaking and I don't have time to look at this now, so if you could please make a separate PR for them, it would be great.

I've added the os-release commit, thanks for the reminder!

swt2c commented:
> Sorry @swt2c I'm getting some weird changes to 3rd party libs (PNG etc) when trying to backport the samples-related stuff and rebaking and I don't have time to look at this now, so if you could please make a separate PR for them, it would be great.

Will do.
